pkg/types/common/common.go

```diff
@@ -0,0 +1,74 @@
+package common
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/go-playground/validator/v10"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// BMC stores the information about a baremetal host's management controller.
+type BMC struct {
+	Username                       string `json:"username" validate:"required"`
+	Password                       string `json:"password" validate:"required"`
+	Address                        string `json:"address" validate:"required,uniqueField"`
+	DisableCertificateVerification bool   `json:"disableCertificateVerification"`
```

Comment:
While I appreciate the desire to unify the common values across our installation option, I am worried that sharing a common struct means that we need to hold both `fencingCredentials.BMC` and `baremetal.Hosts.BMC` in lockstep.

The `DisableCertificateVerification` option is an example of this. Do we plan on building in support for disabling certificate verification for access to the redfish endpoint? If not, then it's confusing if it's an option available in the API. My recommendation would be to keep these as entirely seperate structs, but that they should share conventions when possibles e.g. `Username` `Password` and `Address`

Reply:
>The DisableCertificateVerification option is an example of this

Fair point, I've originally assumed the BMC should be identical.
In case we are using a separate API, I think it would make more sense to drop the `BMC` struct for TNF and integrate it's fields into the  `FencingCredential`, having two `BMC` structs seems confusing to me.
WDYT ?

Reply:
My preference is to keep it totally separate. I'm fine with integrating the BMC fields into FencingCredential. My intuition is that FencingCredential will evolve over time with more options added. I expect BMC to stay static.